File: rps.py
#!/usr/bin/env python3
"""Interactive Rock-Paper-Scissors CLI (human vs. computer).

Run with: python rps.py
"""

import random
import sys

MOVES = ["rock", "paper", "scissors"]
SHORTHAND = {"r": "rock", "p": "paper", "s": "scissors"}
BEATS = {"rock": "scissors", "scissors": "paper", "paper": "rock"}

PROMPT = "Your move (rock/paper/scissors or r/p/s, q to quit): "


def normalize_move(text):
    """Return the canonical move for user input, or None if unrecognized."""
    value = text.strip().lower()
    if value in MOVES:
        return value
    return SHORTHAND.get(value)


def judge(player, computer):
    """Return 'tie', 'win', or 'lose' from the player's perspective."""
    if player == computer:
        return "tie"
    if BEATS[player] == computer:
        return "win"
    return "lose"


def score_line(score):
    return (
        "Score - You: {wins}  Computer: {losses}  Ties: {ties}".format(**score)
    )


def main():
    score = {"wins": 0, "losses": 0, "ties": 0}
    print("Rock-Paper-Scissors! Type 'quit' or 'q' to stop.")
    while True:
        try:
            raw = input(PROMPT)
        except (EOFError, KeyboardInterrupt):
            print()
            break

        command = raw.strip().lower()
        if command in ("quit", "q"):
            break

        player = normalize_move(raw)
        if player is None:
            print("Not recognized. Please enter rock, paper, scissors (or r/p/s).")
            continue

        computer = random.choice(MOVES)
        outcome = judge(player, computer)
        print("You chose {}, computer chose {}.".format(player, computer))
        if outcome == "win":
            score["wins"] += 1
            print("You win!")
        elif outcome == "lose":
            score["losses"] += 1
            print("You lose!")
        else:
            score["ties"] += 1
            print("Tie!")
        print(score_line(score))

    print("Final " + score_line(score))
    print("Thanks for playing!")
    return 0


if __name__ == "__main__":
    sys.exit(main())

File: Rps.java
import java.util.Random;
import java.util.Scanner;

public class Rps {
    private static final String[] MOVES = {"rock", "paper", "scissors"};
    private static final Random random = new Random();

    private static String computerMove() {
        return MOVES[random.nextInt(MOVES.length)];
    }

    private static String outcome(String player, String computer) {
        if (player.equals(computer)) return "TIE";
        if ((player.equals("rock") && computer.equals("scissors"))
                || (player.equals("scissors") && computer.equals("paper"))
                || (player.equals("paper") && computer.equals("rock"))) {
            return "PLAYER_WIN";
        }
        return "COMPUTER_WIN";
    }

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        int playerWins = 0, computerWins = 0, ties = 0;

        while (true) {
            System.out.print("Enter move (rock/paper/scissors) or quit: ");
            if (!scanner.hasNextLine()) break;
            String input = scanner.nextLine().trim().toLowerCase();

            if (input.equals("quit") || input.equals("q")) break;

            boolean valid = false;
            for (String m : MOVES) {
                if (m.equals(input)) { valid = true; break; }
            }
            if (!valid) {
                System.out.println("Invalid input. Please enter rock, paper, scissors, or quit.");
                continue;
            }

            String computer = computerMove();
            String result = outcome(input, computer);
            System.out.println("Computer chose: " + computer);

            if (result.equals("PLAYER_WIN")) {
                playerWins++;
                System.out.println("You win this round!");
            } else if (result.equals("COMPUTER_WIN")) {
                computerWins++;
                System.out.println("Computer wins this round!");
            } else {
                ties++;
                System.out.println("It's a tie!");
            }
            System.out.println("Score — You: " + playerWins + " | Computer: " + computerWins + " | Ties: " + ties);
        }

        System.out.println("\nFinal Score — You: " + playerWins + " | Computer: " + computerWins + " | Ties: " + ties);
        scanner.close();
    }
}
